modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/SubscriptionStore.java

```diff
@@ -447,6 +447,34 @@ public boolean isSubscriptionAvailable(AndesSubscription subscription) throws An
     }
 
     /**
+     * Get ALL (ACTIVE + INACTIVE) local subscriptions whose bound queue is given
+     *
+     * @param queueName Queue name to search
+     * @return List if matching subscriptions
+     * @throws AndesException
+     */
+    public Set<AndesSubscription> getListOfClusterSubscriptionsBoundToQueue(String queueName) throws AndesException {
+        Set<AndesSubscription> subscriptionsOfQueue = new HashSet<>();
+        Set<AndesSubscription> queueSubscriptionMap = clusterQueueSubscriptionMap.get(queueName);
+        if (queueSubscriptionMap != null) {
+            subscriptionsOfQueue.addAll(queueSubscriptionMap);
+        }
+        Map<String, Set<AndesSubscription>> topicSubscriptionMap = clusterTopicSubscriptionMap;
```

Comment:
Do we need to declare a local variable here. Can't we use the clusterTopicSubscriptionMap?